advanced_alchemy/service/typing.py

```diff
@@ -44,16 +44,69 @@
 
 
 FilterTypeT = TypeVar("FilterTypeT", bound="StatementFilter")
+"""Type variable for filter types.
+
+:class:`~advanced_alchemy.filters.StatementFilter`
+"""
 ModelDTOT = TypeVar("ModelDTOT", bound="Struct | BaseModel")
+"""Type variable for model DTOs.
+
+:class:`msgspec.Struct` or :class:`pydantic.BaseModel`
+"""
 PydanticOrMsgspecT = Union[Struct, BaseModel]
+"""Type alias for pydantic or msgspec models.
+
+:class:`msgspec.Struct` or :class:`pydantic.BaseModel`
+"""
 ModelDictT: TypeAlias = Union[Dict[str, Any], ModelT, Struct, BaseModel, DTOData[ModelT]]
+"""Type alias for model dictionaries.
+
+Represents:
+- :type: dict[str, Any]
+- :class:`~advanced_alchemy.base.ModelProtocol`
+- :class:`msgspec.Struct`
+- :class:`pydantic.BaseModel`
+- :class:`litestar.dto.data_structures.DTOData`
+- :class:`~advanced_alchemy.base.ModelProtocol`
+"""
 ModelDictListT: TypeAlias = Sequence[Union[Dict[str, Any], ModelT, Struct, BaseModel]]
-BulkModelDictT: TypeAlias = Union[Sequence[Union[Dict[str, Any], ModelT, Struct, BaseModel]], DTOData[List[ModelT]]]
+"""Type alias for model dictionary lists.
+
+A list or sequence of any of the following:
```

Comment:
I feel this is verbose and is just explaining what the type hint is, I don't have a strong opinion on whether this has to go or stay though.